drivers/clock_control/clock_control_rcar_cpg_mssr.c

```diff
@@ -170,30 +123,13 @@ static int cpg_get_rate(const struct device *dev,
 {
 	const struct rcar_mssr_config *config = dev->config;
 	struct rcar_cpg_clk *clk = (struct rcar_cpg_clk *)sys;
-	uint32_t val;
 	int ret = 0;
 
 	if (clk->domain != CPG_CORE) {
 		return -ENOTSUP;
 	}
 
-	switch (clk->module) {
-	case CPG_CORE_CLK_CANFD:
-		val = sys_read32(config->base_address + CANFDCKCR);
-		if (val & CANFDCKCR_CKSTP) {
-			*rate = 0;
-		} else {
-			val &= CANFDCKCR_DIVIDER_MASK;
-			*rate = CANFDCKCR_PARENT_CLK_RATE / (val + 1);
-		}
-		break;
-	case CPG_CORE_CLK_S3D4:
-		*rate = S3D4_CLK_RATE;
-		break;
-	default:
-		ret = -ENOTSUP;
-		break;
-	}
+	ret = soc_cpg_get_rate(config->base_address, clk->module, rate);
 
 	return ret;
```

Comment:
return soc_cpg_get_rate(config->base_address, clk->module, rate); ?

Reply:
fixed it :+1: